app/models/piece.rb

```diff
@@ -79,7 +79,9 @@ def is_obstructed?(dest_x, dest_y)
     if delta_x != 0 && delta_y != 0 && delta_x.abs != delta_y.abs
       # this handles invalid input or invalid moves.
       # should this include tests for values over 7 and non-numeric input?
-      fail "Invalid input or invalid move."
+      nil
+      # Commented out and replaced with nil: will need to discuss this...
+      # fail "Invalid input or invalid move."
```

Comment:
Discussed this with @amarkpark. Not sure if this will be the final form of this method but with `fail` all my tests will, well, fail :D